### DISCUSSION TITLE
实体类不继承Entity的话 主键又是设置成@TableId(type = IdType.INPUT) 主键怎么赋值？

### DISCUSSION THREAD

**Discussion by zqb666** [zuihou/zuihou-admin-cloud#283] [02.提问]
实体类不继承Entity的话 主键又是设置成@TableId(type = IdType.INPUT) 主键怎么赋值？

**Comment by zuihou** [comment]
名称是id，系统也做了自动赋值。具体看lamp-databases模块里面有一个类

**Reply by zqb666** [reply]
发现 不行 不继承Entity的话 id保存的时候是null

**Reply by zuihou** [reply]
调试 LampMetaObjectHandler

**Reply by zqb666** [reply]
insert的时候断点没进去 没进fillId方法

**Reply by zqb666** [reply]
<img width="796" alt="image" src="https://github.com/dromara/lamp-cloud/assets/6092525/52a92906-ffa9-4ba7-8a84-2f5a45793eb7">
<img width="689" alt="image" src="https://github.com/dromara/lamp-cloud/assets/6092525/bd096694-a9a2-4337-851b-17bbce49286c">
貌似跟注释的不符合

**Reply by zuihou** [reply]
只有1个注解不行，类里面必须有一个create_time、create_by 等任意字段才会进入 LampMetaObjectHandler 类。

原因去问mybaits-plus，他升级版本改了机制。

类里面只有@TableId(type=INPUT) ， 去看mybatis-plus 官方文档，如何注入。

**Reply by zqb666** [reply]
调试了 发现必须 加 
<img width="589" alt="image" src="https://github.com/dromara/lamp-cloud/assets/6092525/92bd6a9c-59ee-47e9-9299-9ad66d084175">
FieldFill.INSERT)这种 cai才能进入到LampMetaObjectHandler 的fillId方法

**Reply by zqb666** [reply]
官网看了 也没什么办法
